packages/synapse-react-client/src/lib/containers/auth/LoginForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect } from 'react'
+import {
+  BackendDestinationEnum,
+  getEndpoint,
+} from '../../utils/functions/getEndpoint'
+import { Box, Link } from '@mui/material'
+import FullWidthAlert from '../FullWidthAlert'
+import { UseLoginReturn } from '../../utils/hooks'
+import { TwoFactorAuthErrorResponse } from '../../utils/synapseTypes/ErrorResponse'
+import TOTPForm from './TOTPForm'
+import UsernamePasswordForm from './UsernamePasswordForm'
+import AuthenticationMethodSelection from './AuthenticationMethodSelection'
+
+type Props = {
+  ssoRedirectUrl?: string
+  registerAccountUrl?: string
+  resetPasswordUrl?: string
+  /* Invoked before redirecting to Google. Useful in portals where we may want to store the current URL to redirect back here. */
+  onBeginOAuthSignIn?: () => void
+  step: UseLoginReturn['step']
+  onStepChange: (step: UseLoginReturn['step']) => void
+  /* optionally pass the 2FA error response to directly start the 2FA challenge */
+  twoFactorAuthenticationRequired?: TwoFactorAuthErrorResponse
+  submitUsernameAndPassword: UseLoginReturn['submitUsernameAndPassword']
+  submitOneTimePassword: UseLoginReturn['submitOneTimePassword']
+  errorMessage: UseLoginReturn['errorMessage']
+}
+
+export default function LoginForm(props: Props) {
```

Comment:
This can be embedded in both StandaloneLoginForm and LoginPage, which need to store the state of the login flow. That state can be passed to this shared component.